### DISCUSSION TITLE
[Android] TextField Input focus issue

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#3228] [Q&A]
Focus behaviour of Input TextField differs from Android native behaviour in a way that on mouseDown TextField gets immediate focus instead of few milliseconds delay. Native Android behavior provides the user with the ability to scroll instead of edit text.
Is there a way to avoid this or would it be possible to add delay feature to TextField?

Here are video example of issue...
[AIR behaviour](https://github.com/airsdk/Adobe-Runtime-Support/assets/56675407/29cbe7ad-48f5-496f-9cc3-b94752f7f790)
[Android tap/click](https://github.com/airsdk/Adobe-Runtime-Support/assets/56675407/46998f89-e447-46ae-95a2-ac34dbaeaa0f)
[Android long press](https://github.com/airsdk/Adobe-Runtime-Support/assets/56675407/d1ea790e-a6d8-47c8-91ee-000e4368cad7)

**Comment by ajwfrost** [comment]
Hi - we've been looking at this a little, am I right in thinking you're using Flex for the display? There may be some interesting interactions between the various frameworks and event handlers to try to sort this behaviour out...

Is it possible we could get the test SWF (and/or source) for this, to try out on a phone here?

thanks

**Reply by MalacTheLittle** [reply]
No Flex, I'm using Adobe Flash CC IDE with AIR.
Sure thing, here's source files... 
[Textfield TEST.zip](https://github.com/user-attachments/files/15922553/Textfield.TEST.zip)

Two more things I noticed:
1. [Android] When pressed on textField and finger moved, `caretIndex` is set to `0`.
2. [Windows] When pressed on textField caretIndex is fine, but moving the mouse will make awkward result which is unconvenient for Windows touch screen. Focus delaying would be welcomed for Windows too. [Video example](https://github.com/airsdk/Adobe-Runtime-Support/assets/56675407/5dc114b5-e883-4066-a578-2003633c9405)

**Comment by ajwfrost** [comment]
Hi @MalacTheLittle  - just wondering, could you please take a look at the behaviour of this on AIR 51.1? We had made a change to the touch handling for text fields but it seems to have not been put into the release notes...

The problem being, it seems to be this change which is causing the problem in #3399 .... so we will have to change it, but I'm curious as to whether the behaviour that's in there now is actually what you were after?

thanks

**Reply by MalacTheLittle** [reply]
This seems to be what I was looking for on Android... just one more thing: the cursor should remain hidden until the intent to edit the text... You can compare these two screenshots:
[AIR 51.1.1.4 behaviour](https://github.com/user-attachments/assets/962e092a-6f7f-441a-aa52-ae17e81db453)
[Android behaviour](https://github.com/airsdk/Adobe-Runtime-Support/assets/56675407/46998f89-e447-46ae-95a2-ac34dbaeaa0f)

Here's the APK too if you want to test it yourself:
[APK for test](https://github.com/user-attachments/files/16985216/TFfocusIssue_NEW.zip)

Still, Windows issue remains:
[Windows behaviour](https://github.com/user-attachments/assets/c691d0e1-26d6-48ce-b1a6-79c8e1be15e5)

**Comment by MalacTheLittle** [comment]
@ajwfrost Does this [revert](https://github.com/airsdk/Adobe-Runtime-Support/discussions/3399#discussioncomment-10634976) you mention mean these changes won't be in future releases?

**Reply by ajwfrost** [reply]
Yes, sorry - I think we need to look at how to implement this better, and we can also try to do this in a more platform-agnostic way to cover the Windows behaviour too..
